duplex_http_call.go

```diff
@@ -236,6 +236,8 @@ func (d *duplexHTTPCall) makeRequest() {
 	response, err := d.httpClient.Do(d.request)
 	if err != nil {
 		err = wrapIfContextError(err)
+		err = wrapIfLikelyH2CNotConfiguredError(err)
```

Comment:
Could we only do this check if the URL starts with "http://"? If it's HTTPS then we definitely don't have h2c problems.

Reply:
Sent the request to the function and check it there, done